drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -384,10 +422,56 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 				clk_buf[i] * SDW_DOUBLE_RATE_FACTOR;
 
 		if (curr_dr_freq * (mstr_prop->default_col - 1) <
-		    bus->params.bandwidth * mstr_prop->default_col)
-			continue;
-
-		break;
+		    bus->params.bandwidth * mstr_prop->default_col) {
+			list_for_each_entry(m_rt, &bus->m_rt_list, bus_node) {
+				/*
+				 * Get the first s_rt that will be used to find the available lane that can be used.
+				 */
+				s_rt = list_first_entry(&m_rt->slave_rt_list, struct sdw_slave_runtime, m_rt_node);
+				slave_prop = &s_rt->slave->prop;
+
+				/*
+				 * Find available manager lanes that connected to the first Peripheral.
+				 * No need to check all Peripherals available lanes because we can't use
+				 * multi-lane if we can't find any available lane for the first Peripheral.
+				 */
+				for (l = 1; l < SDW_MAX_LANES; l++) {
+					if (!slave_prop->lane_maps[l])
+						continue;
+
+					dev_dbg(bus->dev, "%s: trying lane %d\n", __func__, l);
+					required_bandwidth = 0;
+					list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+						required_bandwidth += m_rt->stream->params.rate *
+							hweight32(m_p_rt->ch_mask) *
+							m_rt->stream->params.bps;
+					}
+					if (required_bandwidth <= curr_dr_freq - bus->lane_used_bandwidth[l]) {
+						/* Check if m_lane is connected to all Peripherals */
+						if (!check_all_peripherals_connected(m_rt, slave_prop->lane_maps[l])) {
+							dev_dbg(bus->dev,
+								"some Peripherals are not connected to %d\n",
+								slave_prop->lane_maps[l]);
+							continue;
+						}
+						m_lane = slave_prop->lane_maps[l];
+						dev_dbg(&s_rt->slave->dev,
+							"M lane %d P lane %d can be used\n",
+							m_lane, l);
+						bus->lane_used_bandwidth[l] += required_bandwidth;
+						/*
+						 * Use non-zero manager lane, subtract the lane 0
+						 * bandwidth that is already calculated
+						 */
+						bus->params.bandwidth -= required_bandwidth;
+						use_multi_lane = true;
+						goto out;
+					}
+				}
+			}
+		} else {
+			break;
+		}
```

Comment:
I would revert the logic to do the break first and then you can make the code more readable with one level of indentation removed.

Reply:
Good suggestion. Done.